### PR TITLE
Implemented part of the missing methods from `Parser`

### PR DESCRIPTION
**Pull request by badochov**

### CODE REVIEW

File: src/nodes/nodes.go [hunk 1]
```diff
@@ -12,6 +12,11 @@ type ExprWithName interface {
 	Expr
 }
 
+type StmtWithWithContext interface {
```
Comment (jmolinski) [inline src/nodes/nodes.go]: not very clear name

File: src/parser/parser.go [hunk 2]
```diff
@@ -946,21 +947,15 @@ func (p *parser) parseStatement() ([]nodes.Node, error) {
 			case "autoescape":
 				return wrapInSlice(p.parseAutoescape)
 			default:
-				panic("unexpected statement keyword " + tokenValue)
```
Comment (jmolinski) [inline src/parser/parser.go]: It's really an unexpected thing, it's an error in the parser (not in data feed to parser), i think it should panic

Comment (badochov) [reply src/parser/parser.go]: youre right

**Review by jmolinski — APPROVED** [review]
LGTM

**Pull request merged**